captum/attr/_core/kernel_shap.py

```diff
@@ -29,22 +33,41 @@ def kernel_shap_similarity_kernel(
     ), "Must provide num_interp_features to use default similarity kernel"
     num_selected_features = int(interpretable_sample.sum(dim=1).item())
     num_features = kwargs["num_interp_features"]
-    combinations = combination(num_features, num_selected_features)
-    denom = (
-        combinations * num_selected_features * (num_features - num_selected_features)
-    )
-    if denom != 0:
-        similarities = (num_features - 1) / denom
-    else:
+    if num_selected_features == 0 or num_selected_features == num_features:
         # weight should be theoretically infinite when denom = 0
         # enforcing that trained linear model must satisfy
         # end-point criteria. In practice, it is sufficient to
         # make this weight substantially larger so setting this
-        # weight to 100 (all other weights are < 1).
-        similarities = 100.0
+        # weight to 1000000 (all other weights are 1).
+        similarities = 1000000.0
```

Comment:
Yes, it's a good point that this could need to be customized. For now, to avoid having too many parameters, I can make this an instance variable that advanced users can override on the object after creation, but we can make it a parameter later if necessary.